test/components/guideLineLayerTests.ts

```diff
@@ -1,481 +1,471 @@
-
-      svg.remove();
-    });
-  });
-
-  describe("rendering (horizontal)", () => {
-    let SVG_WIDTH = 300;
-    let SVG_HEIGHT = 400;
-
-    it("requests no space, but will occupy all offered space", () => {
-      let gll = new Plottable.Components.GuideLineLayer<void>("horizontal");
-      let request = gll.requestedSpace(SVG_WIDTH, SVG_HEIGHT);
-      TestMethods.verifySpaceRequest(request, 0, 0, "does not request any space");
-      assert.isTrue(gll.fixedWidth(), "fixed width");
-      assert.isTrue(gll.fixedHeight(), "fixed height");
-
-      let svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
-      gll.anchor(svg);
-      gll.computeLayout({x: 0, y: 0}, SVG_WIDTH, SVG_HEIGHT);
-      assert.strictEqual(gll.width(), SVG_WIDTH, "accepted all offered width");
-      assert.strictEqual(gll.height(), SVG_HEIGHT, "accepted all offered height");
-      svg.remove();
+describe("Interactive Components", () => {
+  describe("GuideLineLayer", () => {
+
+    describe("Basic Usage", () => {
+      it("can get and set the scale property", () => {
+        let gll = new Plottable.Components.GuideLineLayer<Date>("vertical");
+        let timeScale = new Plottable.Scales.Time();
+        assert.isUndefined(gll.scale(), "there is no default scale");
+        assert.strictEqual(gll.scale(timeScale), gll, "setter returns the calling GuideLineLayer");
+        assert.strictEqual(gll.scale(), timeScale, "getter returns the set Scale");
+      });
+
+      it("can get and set the value property", () => {
+        let gll = new Plottable.Components.GuideLineLayer<number>("vertical");
+        let expectedValue = 5;
+        assert.isUndefined(gll.value(), "returns undefined before any value is set");
+        assert.strictEqual(gll.value(expectedValue), gll, "setter returns the calling GuideLineLayer");
+        assert.strictEqual(gll.value(), expectedValue, "getter returns the set value");
+      });
+
+      it("can get and set the pixelPosition property", () => {
+        let gll = new Plottable.Components.GuideLineLayer<void>("vertical");
+        let expectedPosition = 5;
+        assert.isUndefined(gll.pixelPosition(), "returns undefined before any pixel position is set");
+        assert.strictEqual(gll.pixelPosition(expectedPosition), gll, "setter returns the calling GuideLineLayer");
+        assert.strictEqual(gll.pixelPosition(), expectedPosition, "getter returns the set pixel position");
+        // HACKHACK #2614: chai-assert.d.ts has the wrong signature
+        (<any> assert).throws(() => gll.pixelPosition(NaN), Error, "", "Rejects NaN");
+        (<any> assert).throws(() => gll.pixelPosition(Infinity), Error, "", "Rejects Infinity");
+        (<any> assert).throws(() => gll.pixelPosition(-Infinity), Error, "", "Rejects -Infinity");
+        (<any> assert).throws(() => gll.pixelPosition(<any> "5"), Error, "", "Rejects stringy numbers");
```

Comment:
I think the policy now is to include the expected error message when testing with `assert.throws()` right?